src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs

```diff
@@ -227,12 +235,12 @@ public static void Initialize(IPluginCommunication autoRest)
             _ => null
         };
 
-        private static bool GetOptionBoolValue(IPluginCommunication autoRest, string option)
+        private static bool GetOptionBoolValue(IPluginCommunication autoRest, string option, bool isAzureArm)
         {
-            return autoRest.GetValue<bool?>(option).GetAwaiter().GetResult() ?? GetDefaultBoolOptionValue(option)!.Value;
+            return autoRest.GetValue<bool?>(option).GetAwaiter().GetResult() ?? GetDefaultBoolOptionValue(option, isAzureArm)!.Value;
         }
 
-        public static bool? GetDefaultBoolOptionValue(string option)
+        public static bool? GetDefaultBoolOptionValue(string option, bool isAzureArm)
```

Comment:
we already have Options.AzureArm, why do we need this additonal flag?

Reply:
Because this PR adds a new configuration `publicDiscriminatorProperty` which has different default value for mgmt plane and others, therefore I have to add an input here so that we could return different default values.

Reply:
Can we optimize the logic here?  Also seems we only need IsAzureArm for Options.PublicDiscriminatorProperty, but we passed IsAzureArm for all other properties.

Reply:
No that is not an optimization - because potentially we could have different "default values" for all the options, this change enables us to do it. If the default value is the same regardless which plane we are on, that option could just ignore this flag.

Otherwise we have to make another method to give default values and we will have two methods for assigning default values. I will never call that an "optimization".

It is a more valid ask that "why we have to have different default values for mgmt plane and others". I do not like this design as well and I hope we could have a unified default value.

Reply:
Nevertheless, I just pushed a commit that removes this different default value part because I do not like it either.
Now the default value of this configuration is always false. Because it will not be a breaking change if we change it from false to true but it is a breaking change the other way around